cpu/native/netdev_raw802154/netdev_raw802154.c

```diff
@@ -0,0 +1,337 @@
+            native_async_continue_reading(dev->fd);
+        else
+            return 127;
+    }
+
+    nread = real_read(dev->fd, buf, len) - 2;
+    if (nread > 0) {
+        native_async_continue_reading(dev->fd);
+
+#ifdef MODULE_NETSTATS_L2
+        netdev->stats.rx_count++;
+        netdev->stats.rx_bytes += nread;
+#endif
+    }
+
+    return nread;
+}
+
+static int _send(netdev_t *netdev, const iolist_t *iolist)
+{
+    netdev_raw802154_t *dev = (netdev_raw802154_t *)netdev;
+    struct iovec iov[iolist_count(iolist)];
+    size_t bytes;
+    unsigned n;
+
+    bytes = iolist_to_iovec(iolist, iov, &n);
+
+    int res = _native_writev(dev->fd, iov, n);
+#ifdef MODULE_NETSTATS_L2
+    netdev->stats.tx_bytes += bytes;
+#else
+    (void)bytes;
+#endif
+    if (netdev->event_callback)
+        netdev->event_callback(netdev, NETDEV_EVENT_TX_COMPLETE);
+
+    return res;
+}
+
+static int _init(netdev_t *netdev)
+{
+    netdev_raw802154_t *dev = (netdev_raw802154_t *)netdev;
+    uint64_t extended_rand_addr;
+    struct sockaddr_ll sll;
+    struct ifreq ifr;
+    int ret;
+
+#ifdef MODULE_NETSTATS_L2
+    memset(&netdev->stats, 0, sizeof(netstats_t));
+#endif
+
+    /* reset options and sequence number */
+    dev->netdev.seq = 0;
+
+    _set_page(dev, 0);
+    _set_chan(dev, 11);
+
+    if (dev->rand_addr) {
+        hwrng_init();
+        hwrng_read((uint8_t *)&extended_rand_addr, sizeof(extended_rand_addr));
+        /* make sure we mark the address as non-multicast and not globally unique */
+        ((uint8_t *)&extended_rand_addr)[0] &= ~(0x01);
+        ((uint8_t *)&extended_rand_addr)[0] |= 0x02;
+        _set_addr_long(dev, extended_rand_addr);
+    } else {
+        _set_addr_long(dev, RAW802154_DEFAULT_ADDR_LONG);
+    }
+    _set_addr_short(dev, RAW802154_DEFAULT_ADDR_SHORT);
+    _set_pan(dev, RAW802154_DEFAULT_ADDR_PAN);
+
+#ifdef MODULE_GNRC_SIXLOWPAN
+    dev->netdev.proto = GNRC_NETTYPE_SIXLOWPAN;
+#elif MODULE_GNRC
+    dev->netdev.proto = GNRC_NETTYPE_UNDEF;
+#endif
+
+    dev->fd = real_socket(AF_PACKET, SOCK_RAW, htons(ETH_P_IEEE802154));
+    if (dev->fd == -1) {
+        err(EXIT_FAILURE, "Couldn't create a raw 802154 socket\n");
+    }
+
+    strncpy(ifr.ifr_name, dev->ifname, IFNAMSIZ);
+    ret = real_ioctl(dev->fd, SIOCGIFINDEX, &ifr);
+    if (ret < 0) {
+        real_close(dev->fd);
+        err(EXIT_FAILURE, "Failed SIOCGIFINDEX on interface %s", dev->ifname);
+    }
+
+    sll.sll_family = AF_PACKET;
+    sll.sll_ifindex = ifr.ifr_ifindex;
+    sll.sll_protocol = htons(ETH_P_IEEE802154);
+    ret = real_bind(dev->fd, (struct sockaddr *)&sll, sizeof(sll));
+    if (ret == -1) {
+        real_close(dev->fd);
+        err(EXIT_FAILURE, "Failed to bind interface");
+    }
+
+    /* configure signal handler for fds */
+    native_async_read_add_handler(dev->fd, netdev, _raw802154_isr);
+
+    return 0;
+}
+
+void netdev_raw802154_cleanup(netdev_raw802154_t *dev)
+{
+    /* Do we have a device */
+    if (!dev) {
```

Comment:
Is the comment above a question (question mark missing)? Also, I think this can just be an `assert(dev);`.